contracts/protocols/mainnet/mstable/helpers.sol

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+import { DSMath } from "../../../utils/dsmath.sol";
+import "./interfaces.sol";
+
+contract Helpers is DSMath {
+    struct VaultData {
+        uint256 credits;
+        uint256 balance;
+        uint256 exchangeRate;
+        uint256 rewardsEarned;
+        uint256 rewardsUnclaimed;
+        uint256 rewardsLocked;
+    }
+
+    address internal constant mUsdToken = 0xe2f2a5C287993345a840Db3B0845fbC70f5935a5;
+    address internal constant imUsdToken = 0x30647a72Dc82d7Fbb1123EA74716aB8A317Eac19;
+    address internal constant imUsdVault = 0x78BefCa7de27d07DC6e71da295Cc2946681A6c7B;
+
+    /**
+     * @dev Retrieves Reward data from Vault
+     * @notice This Data can be used to calculate rewards that are vested
+     * @param _account address of the account to retrieve Reward data from
+     * @return Reward[] array of Reward data
+     */
+    function getVestingData(address _account) public view returns (Reward[] memory) {
+        uint64 rewardCount = IBoostedSavingsVault(imUsdVault).userData(_account).rewardCount;
+
+        Reward[] memory rewards = new Reward[](rewardCount);
+
+        for (uint256 i = 0; i < rewardCount; i++) {
+            rewards[i] = IBoostedSavingsVault(imUsdVault).userRewards(_account, i);
+        }
+
+        return rewards;
+    }
+
+    /**
+     * @dev Retrieves Reward amounts
+     * @notice Rewards are split up in 3 amounts
+     * @param _account address of the account to retrieve Reward data from
+     * @return earned       => immidiatly available
+     * @return unclaimed    => earned + amounts that came out of vesting
+     * @return locked       => locked in vesting
+     */
+    function getRewards(address _account)
+        public
+        view
+        returns (
+            uint256 earned,
+            uint256 unclaimed,
+            uint256 locked
+        )
+    {
+        // Get rewards data first
+        Reward[] memory rewards = getVestingData(_account);
+
+        earned = IBoostedSavingsVault(imUsdVault).earned(_account);
+
+        (unclaimed, , ) = IBoostedSavingsVault(imUsdVault).unclaimedRewards(_account);
+        locked = 0;
```

Comment:
I think there is no need to initiate it with zero, as solidity already does that.